LDEventSource/LDDataAccumulator.m

```diff
@@ -0,0 +1,31 @@
+//
+//  LDDataAccumulator.m
+//  DarklyEventSource
+//
+//  Created by Mark Pokorny on 5/30/18. +JMJ
+//  Copyright © 2018 Catamorphic Co. All rights reserved.
+//
+
+#import "LDDataAccumulator.h"
+
+@implementation LDDataAccumulator
+-(void)accumulateDataWithString:(NSString*)dataString {
+    if (dataString.length == 0) { return; }
+    if (self.dataString == nil) {
+        self.dataString = dataString;
+        return;
+    }
+    self.dataString = [self.dataString stringByAppendingString:dataString];
+}
+
+-(BOOL)isReadyToParseEvent {
+    if (self.dataString.length == 0) { return NO; }
```

Comment:
Is this fine if `self.dataString == nil`?

Reply:
Yes, if `dataString` is nil, than any function calls will return `0` and this will evaluate to `YES`

Reply:
What a developer-friendly language feature.